### PR TITLE
Filtering spectrum GET requests

### PR DESCRIPTION
**Pull request by guynir42**
Allow users to send queries filtering spectra by various properties, e.g., the object ID (partial match), the origin, label or type, or by the time the spectrum was observed. 

This is useful for groups doing lots of spectroscopy work (like DESI) where finding a list of spectra is important. 
This also obviates the `ObjSpectrumHandler` and `SpectrumRangeHandler`. We can remove them in the future. 

Also allows users to query if a spectrum exists, and get a bit of metadata about it, without pulling the entire wavelength/flux dataset.

### CODE REVIEW

**Comment by guynir42** [comment]
addresses #2748

File: skyportal/handlers/api/source.py [hunk 1]
```diff
@@ -1383,7 +1383,7 @@ class Validator(Schema):
                 comments_filter_author = [comments_filter_author]
             else:
                 return self.error(
-                    "Invalid commentsFilterOrigin value -- must provide at least one string value"
+                    "Invalid commentsFilterAuthor value -- must provide at least one string value"
```
Comment (mcoughlin) [inline skyportal/handlers/api/source.py:1386]: nice

File: skyportal/handlers/api/spectrum.py [hunk 2]
```diff
@@ -165,77 +226,502 @@ def post(self):
...
+              content:
+                application/json:
+                  schema: Error
+        multiple:
+          description: Retrieve multiple spectra with given criteria
+          tags:
+            - spectra
+          parameters:
+            - in: query
+              name: minimal
```
Comment (mcoughlin) [inline skyportal/handlers/api/spectrum.py:257]: maybe minimalMetadata or similar?

File: skyportal/tests/api/test_spectrum.py [hunk 3]
```diff
@@ -9,6 +11,1020 @@
...
+        assert k in minimal_spec  # using multiple spectra, minimal output
+        assert k in full_spec  # using multiple spectra, full output
+        assert k in single_spec  # using single spectra (should be full always)
+        assert minimal_spec[k] == full_spec[k]
+
+    # check that keys of full spec, outside the minimal list, are not included in minimal
+    for k in full_spec.keys():
+        assert k in list_of_keys or k not in minimal_spec
+
+    # make sure full and single are the same
```
Comment (mcoughlin) [inline skyportal/tests/api/test_spectrum.py:1022]: really nice test.

**Review by mcoughlin — APPROVED** [review]
I think minimal should be a bit more verbose name, otherwise really nice.

**Pull request merged**